model_analyzer/monitor/nvml.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from pynvml import *
+from multiprocessing.pool import ThreadPool
+import time
+
+from model_analyzer.monitor.record.memory import MemoryRecord
+from model_analyzer.monitor.record.collector import RecordCollector
+from model_analyzer.monitor import Monitor
+
+
+class NVMLMonitor(Monitor):
+    """Use NVML to monitor GPU metrics
+    """
+
+    def __init__(self, frequency):
+        """
+        Parameters
+        ----------
+        frequency : int
+            Sampling frequency for the metric
+        """
+        super().__init__(frequency)
+        nvmlInit()
+
+        # FIXME Handle the case when the number of GPUs is larger than one and
+        # CUDA_VISIBLE_DEVICES is set
+        self._nvml_handles = []
+        for i, gpu in enumerate(self._gpus):
+            self._nvml_handles.append(nvmlDeviceGetHandleByIndex(i))
+
+        # Is the background thread active
+        self._thread_active = False
+
+        # Background thread collecting results
+        self._thread = None
+
+        # Thread pool
+        self._thread_pool = ThreadPool(processes=1)
```

Comment:
Should this be a variable to this method along with frequency? We may need to track several processes.

Reply:
@dzier What I had in mind is that we create one `Monitor` per process/model that we want to profile. Even if we have multiple processes, we can monitor them using a single background thread. I would probably need to change the `start_recording_metric` APIs to accommodate multiple processes.